constraints/miden-vm/hash.air

```diff
@@ -0,0 +1,156 @@
+mod HashChipletAir
+
+### Constants and periodic columns ################################################################
+
+periodic_columns:
+    k0: [0, 0, 0, 0, 0, 0, 0, 1]
+    k1: [0, 0, 0, 0, 0, 0, 1, 0]
+    k2: [1, 0, 0, 0, 0, 0, 0, 0]
+
+
+### Helper functions/evaluators ###################################################################
+
+# Enforces that column must be binary.
+ev is_binary(main: [a]):
+    enf a^2 - a = 0
+
+
+# Returns binary negation of the value.
+fn binary_not(value: scalar) -> scalar:
+    return 1 - value
```

Comment:
I agree, I put function on the top in the Memory Chiplet for the same reason. Not sure about additional headers though. On the one hand functions and evaluators are very similar, so this separation is not needed. But on the other hand if we have a lot of functions and evaluators, it will be very convenient to have this separation. I'll make this separation for now, let's see how it will look like.